core-services/src/main/java/org/zalando/nakadi/service/SchemaServiceProvider.java

```diff
@@ -0,0 +1,8 @@
+package org.zalando.nakadi.service;
+
+import org.apache.avro.Schema;
+
+public interface SchemaServiceProvider {
```

Comment:
The purpose of this interface is not clear, the `Provider` part indicate that it contains a function like `getSchemaService()` that will return the `SchemaService`. 

Maybe we don't need an interface here, we can directly use `SchemaService` class. But if we do need it, maybe we should name the interface `SchemaService` and the implementations can be things like `EmbededSchemaService`, `PostgresSchemaService`, `ZookeeperSchemaService`, etc. that evolves over time.

Reply:
I used interface here to avoid creating mocks for getting schemas in the current tests. The interface allows me to use AvroSchema.java as implementation for TestSchemaServiceProvider proxy. The purpose of the test is only for tests atm. 

The naming can be improved, I see that it can be confusing with kind of factory style, where one provides the impl by having `get()` method. At the same time, `SchemaServiceProvider` has `getAvroSchema`, which **provides** schemas.

Reply:
Got it, in this case I suggest `SchemaProviderService`.